src/mpid/ch4/shm/posix/posix_coll_gpu_ipc.h

```diff
@@ -270,4 +293,271 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_POSIX_mpi_alltoall_gpu_ipc_read(const void *s
     goto fn_exit;
 }
 
+MPL_STATIC_INLINE_PREFIX int MPIDI_POSIX_mpi_allgather_gpu_ipc_read(const void *sendbuf,
+                                                                    MPI_Aint sendcount,
+                                                                    MPI_Datatype sendtype,
+                                                                    void *recvbuf,
+                                                                    MPI_Aint recvcount,
+                                                                    MPI_Datatype recvtype,
+                                                                    MPIR_Comm * comm_ptr,
+                                                                    MPIR_Errflag_t errflag)
```

Comment:
Should we move these code into `src/mpid/ch4/shm/ipc/coll`?

Reply:
These algorithms are the collectives on the posix level, so I think it should be fine if we leave them in the posix folder. In the meantime, these algorithms use ipc to move the data, so it also makes sense to move them to the ipc folder. I think both should be fine. Which one do you prefer?